### PR TITLE
Matrix3/Matrix4: Refactored .applyToBuffer()

### PR DESCRIPTION
**Pull request by Mugen87**
This PR renames `.applyToBuffer()` to `. applyToBufferAttribute()`. Besides, the methods now support `InterleavedBufferAttribute`.

### CODE REVIEW

File: docs/api/math/Matrix4.html [hunk 1]
```diff
@@ -110,15 +110,13 @@
...
-		of floats in the form [vector1x, vector1y, vector1z, vector2x, vector2y, vector2z, ...], that represent 3D vectors.<br />
-		[page:Number offset] - (optional) index in the array of the first vector's x component. Default is 0.<br />
-		[page:Number length] - (optional) index in the array of the last vector's z component.
-		Default is the last element in the array.<br /><br />
+		[page:BufferAttribute attribute] - An attribute of floats that represent 3D vectors.<br />
+		[page:Number offset] - (optional) index in the attribute of the first vector. Default is 0.<br />
+		[page:Number count] - (optional) index in the attribute of the last vector. Default is the last element in the attribute.<br /><br />
 
-		Multiplies (applies) the upper 3x3 matrix of this matrix to every 3D vector in the [page:ArrayBuffer buffer].
+		Multiplies (applies) the upper 3x3 matrix of this matrix to every 3D vector in the [page:BufferAttribute attribute].
```
Comment (WestLangley) [inline docs/api/math/Matrix4.html]: upper 3x3 ? Also, what is the use case for this method?

File: docs/api/math/Matrix4.html [hunk 2]
```diff
@@ -110,15 +110,13 @@
...
+		<h3>[method:Array applyToBufferAttribute]( [page:BufferAttribute attribute], [page:Number offset], [page:Number count] )</h3>
 		<div>
-		[page:Array buffer] - An [link:https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer ArrayBuffer]
-		of floats in the form [vector1x, vector1y, vector1z, vector2x, vector2y, vector2z, ...], that represent 3D vectors.<br />
-		[page:Number offset] - (optional) index in the array of the first vector's x component. Default is 0.<br />
-		[page:Number length] - (optional) index in the array of the last vector's z component.
-		Default is the last element in the array.<br /><br />
+		[page:BufferAttribute attribute] - An attribute of floats that represent 3D vectors.<br />
+		[page:Number offset] - (optional) index in the attribute of the first vector. Default is 0.<br />
+		[page:Number count] - (optional) index in the attribute of the last vector. Default is the last element in the attribute.<br /><br />
```
Comment (WestLangley) [inline docs/api/math/Matrix4.html]: `count` is a quantity, not an index.

File: docs/api/math/Matrix4.html [hunk 3]
```diff
@@ -110,15 +110,13 @@
...
 
-		<h3>[method:Array applyToBuffer]( [page:ArrayBuffer buffer], [page:Number offset], [page:Number length] )</h3>
+		<h3>[method:Array applyToBufferAttribute]( [page:BufferAttribute attribute], [page:Number offset], [page:Number count] )</h3>
 		<div>
-		[page:Array buffer] - An [link:https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer ArrayBuffer]
-		of floats in the form [vector1x, vector1y, vector1z, vector2x, vector2y, vector2z, ...], that represent 3D vectors.<br />
-		[page:Number offset] - (optional) index in the array of the first vector's x component. Default is 0.<br />
-		[page:Number length] - (optional) index in the array of the last vector's z component.
-		Default is the last element in the array.<br /><br />
+		[page:BufferAttribute attribute] - An attribute of floats that represent 3D vectors.<br />
```
Comment (WestLangley) [inline docs/api/math/Matrix4.html]: This method only makes sense with certain buffer attributes, right? -- even though it is generically-named.

Comment (Mugen87) [inline docs/api/math/Matrix4.html]: I can't think of a better name than `. applyToBufferAttribute()`. Any suggestions?

File: docs/api/math/Matrix4.html (hunk 1 above)
Comment (Mugen87) [inline docs/api/math/Matrix4.html]: I've removed that part of the sentence. The following commit originally added the methods 1b6effb20430811374b648590bd24fa516548641.

**Comment by Mugen87** [comment]
Um, looks like these methods are not used in the repo so far...

**Comment by WestLangley** [comment]
>Um, looks like these methods are not used in the entire repo so far...

Right. What is the use case? And why `offset` and `count`?

**Comment by Mugen87** [comment]
I think the original author added these parameters to keep the interface of `applyToBuffer` similar to `applyToVector3Array`.

**Comment by WestLangley** [comment]
Let's see what others think...

**Comment by mrdoob** [comment]
Feel free to remove what's not needed 👍

**Comment by Mugen87** [comment]
We maybe need these methods if we want to make `BufferGeometry .applyMatrix()` compatible to `InterleavedBufferAttribute` see [here](https://github.com/mrdoob/three.js/blob/dev/src/core/BufferGeometry.js#L123).

**Comment by Mugen87** [comment]
Why not start with this simplified approach? `. applyToBufferAttribute()` without `offset` and `count`. In this way the methods can be used in `BufferGeometry .applyMatrix()`.

**Comment by WestLangley** [comment]
>Why not start with this simplified approach? . applyToBufferAttribute() without offset and count.

That seems reasonable to me. Can you remove `applyToVector3Array()`? There is only one example use case, which can be refactored or inlined if necessary.

**Comment by mrdoob** [comment]
Thanks!

**Comment by unphased** [comment]
Was this really necessary...? Breaking my stuff. I just pasted it back in though and rebuilt though. 

So this was taken out to make the code 0.1% smaller? A noble pursuit, I suppose.

**Comment by mrdoob** [comment]
@unphased Is not only about the size. It's also about the API, documentation, user support, etc. Sorry that I broke your code.

**Pull request merged**